tools/checkpoint_conversion/convert_vit_checkpoints.py

```diff
@@ -0,0 +1,311 @@
+"""Convert ViT checkpoints.
+
+export KAGGLE_USERNAME=XXX
+export KAGGLE_KEY=XXX
+
+python tools/checkpoint_conversion/convert_vit_checkpoints.py \
+    --preset vit_base_patch16_224
+"""
+
+import os
+import shutil
+
+import keras
+import numpy as np
+import torch
+from absl import app
+from absl import flags
+from PIL import Image
+from transformers import ViTForImageClassification
+from transformers import ViTImageProcessor
+
+import keras_hub
+from keras_hub.src.models.vit.vit_backbone import ViTBackbone
+from keras_hub.src.models.vit.vit_image_classifier import ViTImageClassifier
+from keras_hub.src.models.vit.vit_image_classifier_preprocessor import (
+    ViTImageClassifierPreprocessor,
+)
+from keras_hub.src.models.vit.vit_image_converter import ViTImageConverter
+
+FLAGS = flags.FLAGS
+
+PRESET_MAP = {
+    "vit_base_patch16_224": "google/vit-base-patch16-224",
+    "vit_base_patch16_384": "google/vit-base-patch16-384",
+    "vit_large_patch16_224": "google/vit-large-patch16-224",
+    "vit_large_patch16_384": "google/vit-large-patch16-384",
+}
+
+flags.DEFINE_string(
+    "preset",
+    None,
+    f'Must be one of {",".join(PRESET_MAP.keys())}',
+    required=True,
+)
+flags.DEFINE_string(
+    "upload_uri",
+    None,
+    'Could be "kaggle://keras/{variant}/keras/{preset}"',
+    required=False,
+)
+
+
+def convert_model(hf_model):
+    config = hf_model.config.to_dict()
+    image_size = config["image_size"]
+    backbone = ViTBackbone(
+        image_shape=(image_size, image_size, 3),
+        patch_size=config["patch_size"],
+        num_layers=config["num_hidden_layers"],
+        num_heads=config["num_attention_heads"],
+        hidden_dim=config["hidden_size"],
+        mlp_dim=config["intermediate_size"],
+        dropout_rate=config["hidden_dropout_prob"],
+        attention_dropout=config["attention_probs_dropout_prob"],
+        use_mha_bias=config["qkv_bias"],
+    )
+
+    return ViTImageClassifier(
+        backbone=backbone,
+        num_classes=1000,  # num classes in ImageNet
+    )
+
+
+def convert_weights(keras_hub_model, hf_model):
```

Comment:
Could we write this as an in library converter? Seems very doable and then we expose this to anyone wanting to convert a vit checkpoint.

Reply:
https://github.com/keras-team/keras-hub/tree/master/keras_hub/src/utils/transformers